src/Qt/PlayerPrivate.cpp

```diff
@@ -109,15 +109,14 @@ namespace openshot
             const auto render_time = double_micro_sec(time2 - time1);
 
             // Calculate the amount of time to sleep (by subtracting the render time)
-//            auto sleep_time = duration_cast<micro_sec>( (frame_duration - render_time));
-            auto correction_factor = video_frame_diff * video_frame_diff * video_frame_diff * video_frame_diff * video_frame_diff;
-//            auto sleep_time_no_correction = duration_cast<micro_sec>((frame_duration - render_time));
+            auto correction_factor = video_frame_diff * video_frame_diff * video_frame_diff;
             auto sleep_micro_seconds = (frame_duration - render_time) + double_micro_sec (correction_factor);
             auto sleep_time = (sleep_micro_seconds.count() > 0) ? duration_cast<micro_sec>(double_micro_sec(sleep_micro_seconds)) : duration_cast<micro_sec>(double_micro_sec(0.0));
-            if (- (sleep_micro_seconds) > render_time) {
-                skipFrames = floor((-sleep_micro_seconds)/render_time);
-                if (skipFrames * 1.0 > reader->info.fps.ToDouble())
-                    skipFrames = floor(reader->info.fps.ToDouble());
+            if ( - (sleep_micro_seconds.count()) > frame_duration.count()) {
```

Comment:
Let's add some comments to this, to explain what is happening. Also, let's remove the extra `()` and spaces between them.

Reply:
Also, this logic looks very suspicious to me. Why would negative `sleep_micro_seconds` ever be larger than `frame_duration`, which I assume is always positive?

Reply:
Right. I should definitely explain... The idea is if we want to sleep a negative amount of time. Skip `(-sleep_time)/frame_duration` frames

I'll make sure that's clearly explained